packages/experimental/Switch/src/useSwitch.ts

```diff
@@ -6,25 +6,20 @@ import {
   useViewCommandFocus,
   InteractionEvent,
 } from '@fluentui-react-native/interactive-hooks';
-import { SwitchProps, SwitchInfo } from './Switch.types';
-import { AccessibilityState, AccessibilityActionEvent, Platform, LayoutAnimation, UIManager } from 'react-native';
+import { SwitchProps, SwitchInfo, IAnimationConfig } from './Switch.types';
+import { AccessibilityState, AccessibilityActionEvent, Animated, Platform } from 'react-native';
 import { memoize } from '@fluentui-react-native/framework';
 import { useAsToggleWithEvent } from '@fluentui-react-native/interactive-hooks';
 
 const defaultAccessibilityActions = [{ name: 'Toggle' }];
 
-// Use Layout Animation for Knob animating on state change
-const animateSwitchKnob = () => {
-  if (Platform.OS === 'android' && UIManager.setLayoutAnimationEnabledExperimental) {
-    UIManager.setLayoutAnimationEnabledExperimental(true);
-    LayoutAnimation.configureNext(LayoutAnimation.Presets.easeInEaseOut, () => {
-      UIManager.setLayoutAnimationEnabledExperimental(false);
-    });
-  }
-};
+const isMobile = Platform.OS === 'android' || Platform.OS == 'ios';
 
-export const useSwitch = (props: SwitchProps): SwitchInfo => {
+export const useSwitch = (props: SwitchProps, animationConfig?: IAnimationConfig): SwitchInfo => {
   const defaultComponentRef = React.useRef(null);
+  const [animation] = React.useState(new Animated.Value(0));
```

Comment:
Wouldn't you need to be able to change the value of the animation state at some point? Does the Animation API do that for you?

The example on the RN website uses refs to track the current state: https://reactnative.dev/docs/animated

Reply:
animation value is changed using the apis such as interpolate or timing provided by AnimatedAPI , setting up toValue property. 
I think it's better to use useRef for now then.